plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -421,14 +486,34 @@ public function get_views() {
 	 * @return int
 	 */
 	private function count_orders_by_status( $status ): int {
-		return array_sum(
-			array_map(
-				function( $order_status ) {
-					return wc_orders_count( $order_status, 'shop_order' );
-				},
-				(array) $status
+		global $wpdb;
+
+		$orders_table = OrdersTableDataStore::get_orders_table_name();
+		$status       = (array) $status;
+
+		$status_placeholders = implode( ',', array_fill( 0, count( $status ), '%s' ) );
+
+		$count = absint(
```

Comment:
we can improve this query by using a group by here and doing it for all statuses at once instead of firing this query again and again. since this is a private method, it should be fine to change the signature, wdyt? as far as I can see, a group by query on status should be similarly fast to a normal count query for a single status (since MYSQL would need to scan entire `status` index anyway) making us overall N times faster where N is number of status.

Reply:
I didn't actually change the signature, but the first time the function is called it computes counts for all statuses and then caches that. It's implemented in b98a1660b4de8fee24082937a986568455e1f72f. Let me know what you think 😃